plugins/modules/k8s_cp.py

```diff
@@ -0,0 +1,385 @@
+            msg = "Namespace={0} Kind=Pod Name={1}: Failed requested object: {2}".format(namespace, name, exc.body)
+        else:
+            msg = to_native(exc)
+        for attr in ['status', 'reason']:
+            if hasattr(exc, attr):
+                arg[attr] = getattr(exc, attr)
+        module.fail_json(msg=msg, **arg)
+
+    try:
+        result = resource.get(name=name, namespace=namespace)
+        containers = [c['name'] for c in result.to_dict()['status']['containerStatuses']]
+        if container and container not in containers:
+            module.fail_json(msg="Pod '{0}' from Namespace '{1}' has no container named '{2}'.".format(
+                name, namespace, container
+            ))
+        return containers
+    except Exception as exc:
+        _fail(exc)
+
+
+class K8SCopyModule(object):
+
+    def __init__(self, module, client):
+        self.client = client
+        self.module = module
+        self.api_instance = core_v1_api.CoreV1Api(client.client)
+
+        self.local_path = module.params.get('local_path')
+        self.name = module.params.get('pod')
+        self.namespace = module.params.get('namespace')
+        self.remote_path = module.params.get('remote_path')
+        self.content = module.params.get('content')
+
+        self.container_arg = {}
+        if module.params.get('container'):
+            self.container_arg['container'] = module.params.get('container')
+
+    def test_remote_path(self, as_dir=True):
+        """
+            When 'as_dir' is set to True, we check if the resource exist as a directory inside the remote pod
+            When 'as_dir' is set to False, we check if the resource exist as a file inside the remote pod
+        """
+        try:
+            test_arg = '-d' if as_dir else '-f'
+            test_cmd = ['test', test_arg, self.remote_path]
+            resp = stream(self.api_instance.connect_get_namespaced_pod_exec,
+                          self.name,
+                          self.namespace,
+                          command=test_cmd,
+                          stdout=True, stderr=True,
+                          stdin=False, tty=False,
+                          _preload_content=False, **self.container_arg)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to execute on pod {0}/{1} due to : {2}".format(self.namespace, self.name, to_native(e)))
+        stdout, stderr = [], []
+        while resp.is_open():
+            resp.update(timeout=1)
+            if resp.peek_stdout():
+                stdout.append(resp.read_stdout())
+            if resp.peek_stderr():
+                stderr.append(resp.read_stderr())
+        err = resp.read_channel(3)
+        err = yaml.safe_load(err)
+        resp.close()
+        if err['status'] == 'Success':
+            return True
+        return False
+
+    def copy_local_file_to_remote(self, local_file, remote_file):
+        tar_command = ['tar', 'xvf', '-', '-C', '/']
```

Comment:
```suggestion
        from ansible.module_utils.common.process import get_bin_path
        try:
            tar_bin = get_bin_path('tar')
        except ValueError:
        	self.module.fail_json("Unable to find the 'tar' binary in the path.")
        tar_command = [tar_bin, 'xvf', '-', '-C', '/']
```

Reply:
This is the command which will be executed on the kubernetes stream, so the tar executable is the one from the pod, so retrieving the bin path from the local host is not what we intend.